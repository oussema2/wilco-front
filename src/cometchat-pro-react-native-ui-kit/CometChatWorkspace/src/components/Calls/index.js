export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
