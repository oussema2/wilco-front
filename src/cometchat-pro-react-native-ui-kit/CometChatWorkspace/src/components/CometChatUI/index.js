/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
