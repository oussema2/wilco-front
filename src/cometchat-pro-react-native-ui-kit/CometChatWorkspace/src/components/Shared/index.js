export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
