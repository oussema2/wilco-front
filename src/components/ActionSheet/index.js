export { default as ActionSheet } from './ActionSheet';
