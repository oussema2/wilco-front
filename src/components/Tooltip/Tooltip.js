import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { Tooltip as TooltipElements } from 'react-native-elements';
import { observer } from 'mobx-react-lite';
import { palette } from '../../Theme';
import { styles } from './Tooltip.styles';
import noop from '../../helpers/noop';

const Tooltip = observer( ( {
	testID, children, toggleOnPress, popover, width, withOverlay, onClose
}, ref ) => (
	<View
		testID={testID}
		style={styles.container}
	>
		<TooltipElements
			ref={ref}
			width={width}
			height={null}
			toggleOnPress={toggleOnPress}
			backgroundColor={palette.secondary.default}
			containerStyle={styles.tooltipContainer}
			popover={popover}
			withOverlay={withOverlay}
			onClose={onClose}
		>
			{children}
		</TooltipElements>
	</View>
), { forwardRef: true } );

Tooltip.propTypes = {
	testID: PropTypes.string,
	children: PropTypes.node,
	toggleOnPress: PropTypes.bool,
	popover: PropTypes.node,
	width: PropTypes.number,
	withOverlay: PropTypes.bool,
	onClose: PropTypes.func
};

Tooltip.defaultProps = {
	testID: 'comment-Component',
	children: null,
	toggleOnPress: true,
	popover: null,
	width: 246,
	withOverlay: true,
	onClose: noop
};

export default Tooltip;
