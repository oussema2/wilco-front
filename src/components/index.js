export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
