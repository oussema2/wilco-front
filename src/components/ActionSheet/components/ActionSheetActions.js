import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
