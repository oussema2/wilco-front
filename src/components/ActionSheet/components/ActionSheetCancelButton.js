import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
