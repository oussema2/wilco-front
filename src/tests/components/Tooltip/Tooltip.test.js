import React from 'react';
import { render } from '@testing-library/react-native';
import { Tooltip } from '../../../components/Tooltip';

describe( '<Tooltip />', () => {
	let component;
	const testID = 'tooltip-Component';

	const setUp = ( props ) => {
		component = render( <Tooltip
			testID={testID}
			{...props}
		/> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the Tooltip component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
