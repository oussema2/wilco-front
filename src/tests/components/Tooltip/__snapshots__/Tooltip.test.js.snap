// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> with the default props renders the Tooltip component correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
    }
  }
  testID="tooltip-Component"
>
  <View
    collapsable={false}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onShow={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(250, 250, 250, 0.70)",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 0,
                "left": 0,
                "overflow": "visible",
                "position": "absolute",
                "top": 0,
                "width": 0,
              }
            }
          />
          <View
            style={
              Object {
                "left": -7.5,
                "position": "absolute",
                "top": -2,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "#1F2E4E",
                  "borderBottomWidth": 15,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 8,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 8,
                  "borderStyle": "solid",
                  "height": 0,
                  "width": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1F2E4E",
                "borderRadius": 10,
                "display": "flex",
                "flex": 1,
                "height": null,
                "justifyContent": "center",
                "left": -17.999,
                "padding": 16,
                "position": "absolute",
                "top": 10,
                "width": 246,
              }
            }
            testID="tooltipPopoverContainer"
          />
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
